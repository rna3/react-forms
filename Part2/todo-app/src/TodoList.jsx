import React, { useState } from 'react';
import { v4 as uuid } from "uuid";
import NewTodoForm from './NewTodoForm';
import Todo from './Todo';

function TodoList() {
  const [todos, setTodos] = useState([]);

  const addTodo = (task) => {
    setTodos([...todos, { id: uuid(), task }]);
  };

  const removeTodo = (id) => {
    setTodos(todos.filter(todo => todo.id !== id));
  };

  return (
    <div>
      <NewTodoForm addTodo={addTodo} />
      {todos.map(todo => (
        <Todo key={todo.id} id={todo.id} task={todo.task} removeTodo={removeTodo} />
      ))}
    </div>
  );
}

export default TodoList;
