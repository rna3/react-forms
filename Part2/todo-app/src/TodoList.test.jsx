import React from 'react';
import { render } from '@testing-library/react';
import TodoList from './TodoList';

test("renders TodoList without crashing", () => {
  render(<TodoList />);
});

test("matches snapshot for TodoList", () => {
    const { asFragment } = render(<TodoList />);
    expect(asFragment()).toMatchSnapshot();
  });