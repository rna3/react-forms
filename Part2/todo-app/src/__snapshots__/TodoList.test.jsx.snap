// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot for TodoList 1`] = `
<DocumentFragment>
  <div>
    <form>
      <input
        placeholder="New task"
        type="text"
        value=""
      />
      <button>
        Add
      </button>
    </form>
  </div>
</DocumentFragment>
`;
